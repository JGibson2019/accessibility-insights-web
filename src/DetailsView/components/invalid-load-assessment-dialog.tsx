// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { Dialog, DialogFooter, DialogType, PrimaryButton } from 'office-ui-fabric-react';
import * as styles from 'DetailsView/components/invalid-load-assessment-dialog.scss';
import * as React from 'react';

export interface InvalidLoadAssessmentDialogProps {
    isOpen: boolean;
    onClose: () => void;
}

export const InvalidLoadAssessmentDialog = NamedFC<InvalidLoadAssessmentDialogProps>(
    'InvalidLoadAssessmentDialog',
    props => {
        const onDismiss = (): void => {
            props.onClose();
        };
        if (!props.isOpen) {
            return null;
        }

        return (
            <Dialog
                hidden={false}
                //JG note: clean this up and sub in onClose if we don't have something to add on line 15
                onDismiss={onDismiss}
                dialogContentProps={{
                    type: DialogType.normal,
                    title: 'Oops! Accessibility Insights canâ€™t open this file.',
                    subText: 'Please select a valid *.a11yAssessment file',
                }}
                modalProps={{
                    isBlocking: false,
                    containerClassName: styles.invalidLoadAssessmentDialog,
                    onDismissed: props.onClose,
                }}
            >
                <DialogFooter>
                    <PrimaryButton onClick={props.onClose}>OK</PrimaryButton>
                </DialogFooter>
            </Dialog>
        );
    },
);
