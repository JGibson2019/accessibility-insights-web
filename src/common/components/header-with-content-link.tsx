// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as styles from 'common/components/header-with-content-common.scss';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import { ContentLink, ContentLinkDeps } from 'views/content/content-link';
import { ContentReference } from 'views/content/content-page';

export interface HeaderWithContentLinkProps {
    deps: ContentLinkDeps;
    headerClass?: string;
    headingTitleClassName?: string;
    headingTitle: string;
    secondaryText?: string;
    guidance?: ContentReference;
}

export const HeaderWithContentLink = NamedFC<HeaderWithContentLinkProps>(
    'HeaderWithContent Link',
    props => {
        return (
            <div className={`${styles.headerWithContentLink} ${props.headerClass}`}>
                <h1>
                    <span
                        className={props.headingTitleClassName}
                    >{`${props.headingTitle} ${props.secondaryText}`}</span>
                </h1>
                <ContentLink deps={props.deps} reference={props.guidance} iconName="info" />
            </div>
        );
    },
);
