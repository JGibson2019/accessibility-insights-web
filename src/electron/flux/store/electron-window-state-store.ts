// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseStoreImpl } from 'background/stores/base-store-impl';
import { StoreNames } from 'common/stores/store-names';
import { WindowStateActions } from '../action/window-state-actions';
import { ElectronWindowStateStoreData } from '../types/electron-window-state-store-data';

export class ElectronWindowStateStore extends BaseStoreImpl<ElectronWindowStateStoreData> {
    constructor(private readonly windowStateActions: WindowStateActions) {
        super(StoreNames.ElectronWindowStateStore);
    }

    public getDefaultState(): ElectronWindowStateStoreData {
        return this.getDeviceConnectWindowState();
    }

    protected addActionListeners(): void {
        this.windowStateActions.setDeviceConnectRoute.addListener(this.onSetDeviceConnectRoute);
        this.windowStateActions.setResultsViewRoute.addListener(this.onSetResultsViewRoute);
    }

    private getDeviceConnectWindowState(): ElectronWindowStateStoreData {
        return {
            routeId: 'deviceConnectView',
            windowWidth: 600,
            windowHeight: 391,
        };
    }

    private getResultViewWindowState(): ElectronWindowStateStoreData {
        return {
            routeId: 'resultsView',
            windowWidth: 1366,
            windowHeight: 768,
        };
    }

    private onSetDeviceConnectRoute = () => {
        this.state = this.getDeviceConnectWindowState();
    };

    private onSetResultsViewRoute = () => {
        this.state = this.getResultViewWindowState();
    };
}
