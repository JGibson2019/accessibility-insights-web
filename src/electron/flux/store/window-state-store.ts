// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseStoreImpl } from 'background/stores/base-store-impl';
import { StoreNames } from 'common/stores/store-names';
import { WindowStateActions } from '../action/window-state-actions';
import { WindowStateStoreData } from '../types/window-state-store-data';

export class WindowStateStore extends BaseStoreImpl<WindowStateStoreData> {
    constructor(private readonly windowStateActions: WindowStateActions) {
        super(StoreNames.WindowStateStore);
    }

    public getDefaultState(): WindowStateStoreData {
        return this.getDeviceConnectWindowState();
    }

    protected addActionListeners(): void {
        this.windowStateActions.setDeviceConnectRoute.addListener(this.onSetDeviceConnectRoute);
        this.windowStateActions.setResultsViewRoute.addListener(this.onSetResultsViewRoute);
    }

    private getDeviceConnectWindowState(): WindowStateStoreData {
        return {
            routeId: 'deviceConnectView',
        };
    }

    private getResultViewWindowState(): WindowStateStoreData {
        return {
            routeId: 'resultsView',
        };
    }

    private onSetDeviceConnectRoute = () => {
        this.state = this.getDeviceConnectWindowState();
        this.emitChanged();
    };

    private onSetResultsViewRoute = () => {
        this.state = this.getResultViewWindowState();
        this.emitChanged();
    };
}
