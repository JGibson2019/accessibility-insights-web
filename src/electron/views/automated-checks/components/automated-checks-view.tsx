// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

export type AutomatedChecksViewDeps = {};

export type AutomatedChecksViewProps = {
    deps: AutomatedChecksViewDeps;
};

export class AutomatedChecksView extends React.Component<AutomatedChecksViewProps> {
    public render(): JSX.Element {
        return <>automated checks</>;
    }
}
