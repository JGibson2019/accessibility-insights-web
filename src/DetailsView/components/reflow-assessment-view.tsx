// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDefaultMessageGenerator } from 'assessments/assessment-default-message-generator';
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { AssessmentViewUpdateHandler } from 'DetailsView/components/assessment-view-update-handler';
import * as styles from 'DetailsView/components/assessment-view.scss';
import * as React from 'react';
import { ContentLink, ContentLinkDeps } from 'views/content/content-link';
import { ContentPageComponent } from 'views/content/content-page';

import { AssessmentTestResult } from '../../common/assessment/assessment-test-result';
import { CollapsibleComponent } from '../../common/components/collapsible-component';
import { reactExtensionPoint } from '../../common/extensibility/react-extension-point';
import { Tab } from '../../common/itab';
import {
    AssessmentData,
    AssessmentNavState,
    PersistedTabInfo,
} from '../../common/types/store-data/assessment-result-data';
import { FeatureFlagStoreData } from '../../common/types/store-data/feature-flag-store-data';
import { PathSnippetStoreData } from '../../common/types/store-data/path-snippet-store-data';
import { DetailsViewActionMessageCreator } from '../actions/details-view-action-message-creator';
import { DetailsViewExtensionPoint } from '../extensions/details-view-extension-point';
import { AssessmentInstanceTableHandler } from '../handlers/assessment-instance-table-handler';
import { GettingStartedView, GettingStartedViewProps } from './getting-started-view';
import { TargetChangeDialog, TargetChangeDialogDeps } from './target-change-dialog';
import { TestStepView, TestStepViewDeps } from './test-step-view';
import { TestStepNavDeps, TestStepsNav } from './test-steps-nav';

export type WithAssessmentTestResult = { assessmentTestResult: AssessmentTestResult };
export const AssessmentViewMainContentExtensionPoint = reactExtensionPoint<
    WithAssessmentTestResult
>('AssessmentViewMainContentExtensionPoint');

export type AssessmentViewDeps = ContentLinkDeps &
    TestStepViewDeps &
    TestStepNavDeps &
    TargetChangeDialogDeps & {
        detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
        assessmentsProvider: AssessmentsProvider;
        assessmentViewUpdateHandler: AssessmentViewUpdateHandler;
        detailsViewExtensionPoint: DetailsViewExtensionPoint;
    };

export interface AssessmentViewProps {
    deps: AssessmentViewDeps;
    isScanning: boolean;
    isEnabled: boolean;
    assessmentNavState: AssessmentNavState;
    assessmentInstanceTableHandler: AssessmentInstanceTableHandler;
    assessmentData: AssessmentData;
    currentTarget: Tab;
    prevTarget: PersistedTabInfo;
    assessmentDefaultMessageGenerator: AssessmentDefaultMessageGenerator;
    assessmentTestResult: AssessmentTestResult;
    featureFlagStoreData: FeatureFlagStoreData;
    pathSnippetStoreData: PathSnippetStoreData;
}

export class ReflowAssessmentView extends React.Component<AssessmentViewProps> {
    public static readonly requirementsTitle: string = 'Requirements';

    private deps: AssessmentViewDeps;

    constructor(props: AssessmentViewProps) {
        super(props);
        this.deps = props.deps;
        this.state = {
            isStartOverDialogOpen: false,
        };
    }

    public render(): JSX.Element {
        const { assessmentTestResult } = this.props;
        const { extensions } = assessmentTestResult.definition;
        const extPointProps = { extensions, assessmentTestResult };
        return (
            <div className={styles.assessmentContent}>
                {this.renderTargetChangeDialog()}
                {this.renderTitle(
                    assessmentTestResult.definition.title,
                    assessmentTestResult.definition.guidance,
                )}
                {this.renderGettingStarted(assessmentTestResult.definition.gettingStarted)}
                <AssessmentViewMainContentExtensionPoint.component {...extPointProps}>
                    {this.renderRequirements()}
                    {this.renderMainContent(assessmentTestResult)}
                </AssessmentViewMainContentExtensionPoint.component>
            </div>
        );
    }

    public componentDidMount(): void {
        this.deps.assessmentViewUpdateHandler.onMount(this.props);
    }

    public componentDidUpdate(prevProps: AssessmentViewProps): void {
        this.deps.assessmentViewUpdateHandler.update(prevProps, this.props);

        const { assessmentTestResult } = this.props;
        this.deps.detailsViewExtensionPoint
            .apply(assessmentTestResult.definition.extensions)
            .onAssessmentViewUpdate(prevProps, this.props);
    }

    public componentWillUnmount(): void {
        this.deps.assessmentViewUpdateHandler.onUnmount(this.props);
    }

    private renderTargetChangeDialog(): JSX.Element {
        return (
            <TargetChangeDialog
                deps={this.props.deps}
                prevTab={this.props.prevTarget}
                newTab={this.props.currentTarget}
            />
        );
    }

    private renderTitle(title: string, content?: ContentPageComponent): JSX.Element {
        return (
            <div className={styles.assessmentTitle}>
                <h1 className={styles.assessmentHeader}>
                    {title} <ContentLink deps={this.deps} reference={content} iconName="info" />
                </h1>
            </div>
        );
    }

    private renderGettingStarted(gettingStarted: JSX.Element): JSX.Element {
        return (
            <CollapsibleComponent
                header={<h2>Getting Started</h2>}
                content={gettingStarted}
                containerClassName={styles.assessmentGettingStartedContainer}
            />
        );
    }

    private renderRequirements(): JSX.Element {
        return (
            <div className={styles.assessmentRequirementsTitle}>
                <h2 className={styles.assessmentRequirements}>
                    {ReflowAssessmentView.requirementsTitle}
                </h2>
            </div>
        );
    }

    private renderMainContent(assessmentTestResult: AssessmentTestResult): JSX.Element {
        if (this.props.assessmentNavState.selectedTestSubview === 'getting-started') {
            const props: GettingStartedViewProps = {
                gettingStartedContent: (
                    <div>
                        Could pass the getting started content for the requirement as part of props
                    </div>
                ),
            };
            return (
                <div className={styles.detailsViewAssessmentContent}>
                    <GettingStartedView {...props} />
                </div>
            );
        } else {
            const selectedRequirement = assessmentTestResult.getRequirementResult(
                this.props.assessmentNavState.selectedTestSubview,
            );
            const isStepScanned = selectedRequirement.data.isStepScanned;
            return (
                <div className={styles.detailsViewAssessmentContent}>
                    <div className={styles.testStepsNavContainer}>
                        <TestStepsNav
                            deps={this.props.deps}
                            ariaLabel={ReflowAssessmentView.requirementsTitle}
                            selectedTest={assessmentTestResult.visualizationType}
                            selectedTestStep={selectedRequirement.definition.key}
                            stepStatus={this.props.assessmentData.testStepStatus}
                        />
                    </div>
                    <div className={styles.testStepViewContainer}>
                        <TestStepView
                            deps={this.deps}
                            isScanning={this.props.isScanning}
                            testStep={selectedRequirement.definition}
                            renderStaticContent={selectedRequirement.definition.renderStaticContent}
                            instancesMap={this.props.assessmentData.generatedAssessmentInstancesMap}
                            assessmentNavState={this.props.assessmentNavState}
                            assessmentInstanceTableHandler={
                                this.props.assessmentInstanceTableHandler
                            }
                            manualTestStepResultMap={
                                this.props.assessmentData.manualTestStepResultMap
                            }
                            assessmentsProvider={this.props.deps.assessmentsProvider}
                            isStepEnabled={this.props.isEnabled}
                            isStepScanned={isStepScanned}
                            assessmentDefaultMessageGenerator={
                                this.props.assessmentDefaultMessageGenerator
                            }
                            featureFlagStoreData={this.props.featureFlagStoreData}
                            pathSnippetStoreData={this.props.pathSnippetStoreData}
                        />
                    </div>
                </div>
            );
        }
    }
}
