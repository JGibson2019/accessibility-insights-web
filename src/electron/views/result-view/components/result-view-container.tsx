import { WindowStateStore } from 'electron/flux/store/window-state-store';
import { ClientStoresHub } from 'common/stores/client-stores-hub';

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export type ResultViewContainerDeps = {
    storeHub: ClientStoresHub<WindowStateStore>;
};

export type ResultViewContainerProps = {
    deps: ResultViewContainerDeps;
};
