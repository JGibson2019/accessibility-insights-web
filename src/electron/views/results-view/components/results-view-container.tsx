// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import { ClientStoresHub } from 'common/stores/client-stores-hub';
import { WindowStateStore } from 'electron/flux/store/window-state-store';

export type ResultsViewContainerDeps = {};

export type ResultsViewContainerProps = {
    deps: ResultsViewContainerDeps;
};

export type ResultViewContainerState = {};

export class ResultsViewContainer extends React.Component<ResultsViewContainerProps, ResultViewContainerState> {
    constructor(props: ResultsViewContainerProps) {
        super(props);
    }

    public render(): JSX.Element {
        return null;
    }
}
