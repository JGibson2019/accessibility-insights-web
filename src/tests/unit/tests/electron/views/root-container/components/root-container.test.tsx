// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import { BaseClientStoresHub } from 'common/stores/base-client-stores-hub';
import { ClientStoresHub } from 'common/stores/client-stores-hub';
import { WindowStateStore } from 'electron/flux/store/window-state-store';
import { WindowStateStoreData } from 'electron/flux/types/window-state-store-data';
import { DeviceConnectViewContainerState } from 'electron/views/device-connect-view/components/device-connect-view-container';
import {
    RootContainer,
    RootContainerDeps,
    RootContainerProps,
    RootContainerState,
} from 'electron/views/root-container/components/root-container';
import { shallow } from 'enzyme';
import { It, Mock } from 'typemoq';

describe(RootContainer, () => {
    let storeListener: () => Promise<void>;
    let deps: RootContainerDeps;
    let windowStateStoreData: WindowStateStoreData;

    beforeEach(() => {
        windowStateStoreData = {
            routeId: 'deviceConnectView',
        };

        const storeHubMock = Mock.ofType<
            ClientStoresHub<DeviceConnectViewContainerState> & ClientStoresHub<WindowStateStore> & ClientStoresHub<RootContainerState>
        >(BaseClientStoresHub as any);
        storeHubMock
            .setup(hub => hub.getAllStoreData())
            .returns(() => {
                return { windowStateStoreData } as any;
            });
        storeHubMock
            .setup(w => w.addChangedListenerToAllStores(It.isAny()))
            .callback(cb => {
                storeListener = cb;
            });

        deps = { storeHub: storeHubMock.object } as any;
    });

    it('renders default view from default route', async () => {
        const props: RootContainerProps = { deps };
        const wrapped = shallow(<RootContainer {...props} />);

        wrapped.instance();
        windowStateStoreData.routeId = 'deviceConnectView';

        expect(wrapped.getElement()).toMatchSnapshot('default view');
    });

    it('renders device connect view container when route is deviceConnectView', async () => {
        const props: RootContainerProps = { deps };
        const wrapped = shallow(<RootContainer {...props} />);

        wrapped.instance();
        windowStateStoreData.routeId = 'deviceConnectView';
        await storeListener();

        expect(wrapped.getElement()).toMatchSnapshot('device connect view');
    });

    it('renders results view container when route is resultsView', async () => {
        const props: RootContainerProps = { deps };
        const wrapped = shallow(<RootContainer {...props} />);

        wrapped.instance();
        windowStateStoreData.routeId = 'resultsView';
        await storeListener();

        expect(wrapped.getElement()).toMatchSnapshot('results view');
    });
});
