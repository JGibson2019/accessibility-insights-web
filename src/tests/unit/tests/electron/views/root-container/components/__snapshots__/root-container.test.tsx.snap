// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootContainer renders default view from default route: default view 1`] = `
<DeviceConnectViewContainer
  deps={
    Object {
      "storeHub": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "stores": undefined,
      },
    }
  }
/>
`;

exports[`RootContainer renders device connect view container when route is deviceConnectView: device connect view 1`] = `
<DeviceConnectViewContainer
  deps={
    Object {
      "storeHub": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "stores": undefined,
      },
    }
  }
/>
`;

exports[`RootContainer renders results view container when route is resultsView: results view 1`] = `
<AutomatedChecksView
  deps={
    Object {
      "storeHub": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "stores": undefined,
      },
    }
  }
/>
`;
